### ISSUE TITLE
Internvl-g image-text maching issues

### ISSUE THREAD

**Issue by ljqnb** [OpenGVLab/InternVL#59]
###============== Image-Text Contrastive ===================###
        image_itc = self.clip_projector2(image_itc)

        selected = summarize_attention_mask.sum(1) - 1
        text_itc = text_itc[torch.arange(text_itc.shape[0]), selected]
        text_itc = text_itc @ self.text_projection
你好这个地方是不是有点问题，还是我理解错了，我看原文章写的是取 eos token 的 feature，哪应该是text_itc = text_itc[torch.arange(text_itc.shape[0]), -1]? 因为 llama tokenizer 是左 padding 的。谢谢作者

**Comment by ljqnb** [comment]
我看了一遍源码发现是 right padding，打扰了
